src/test/java/hudson/plugins/promoted_builds/integrations/jobdsl/PromotionsDslContextExtensionTest.java

```diff
@@ -99,4 +99,20 @@ public void testShouldGenerateTheJobWithBuildWrappers() throws Exception {
                 .matcher(content).find());
     }
 
+    @Test
+    public void testShouldGenerateTheDynamicDslJob() throws Exception {
+        // Given
+        String dsl = FileUtils.readFileToString(new File("src/test/resources/dynamic-dsl-example-dsl.groovy"));
+        FreeStyleProject seedJob = j.createFreeStyleProject();
+        seedJob.getBuildersList().add(createScript(dsl));
+        // When
+        QueueTaskFuture<FreeStyleBuild> scheduleBuild2 = seedJob.scheduleBuild2(0);
+        j.assertBuildStatusSuccess(scheduleBuild2.get());
+
+        TopLevelItem item = j.jenkins.getItem("dynamic-dsl-test");
+        File config = new File(item.getRootDir(), "promotions/Development/config.xml");
+        String content = Files.readString(config.toPath());
+        assert content.contains("<javaposse.jobdsl.plugin.ExecuteDslScripts");
```

Comment:
Damn merged to early!

Generally please avoid using an `assert` for a non boolean primative.

If things fail, you just get a failure,  rather it is better to use a higher level framework such as hamcrest so that when things fail you are told that "the needle does not exist in the haystack" (as the haystack can lead you directly to the cause/fix)